shell/src/test/java/alluxio/cli/fsadmin/command/SummaryCommandTest.java

```diff
@@ -0,0 +1,187 @@
+  private static final long START_TIME_MS = 213124234312231L;
+  private static final long UPTIME_MS = 12412412312L;
+  private static final String VERSION = "testVersion";
+  private static final boolean SAFE_MODE = false;
+  private static final int LIVE_WORKER_NUM = 12;
+  private static final int LOST_WORKER_NUM = 4;
+  private static final long CAPACITY_BYTES = 1341353L;
+  private static final long USED_BYTES = 62434L;
+  private static final long FREE_BYTES = 1278919L;
+
+  private MetaMasterClient mMetaMasterClient;
+  private BlockMasterClient mBlockMasterClient;
+  private ByteArrayOutputStream mOutputStream;
+  private PrintStream mPrintStream;
+  private int mIndentationLevel = 1;
+  private Map<String, Long> mCapacityBytesOnTiers = new HashMap<>();
+  private Map<String, Long> mUsedBytesOnTiers = new HashMap<>();
+
+  @Before
+  public void prepareDependencies() throws IOException {
+    // Prepare mock meta master client
+    mMetaMasterClient = Mockito.mock(MetaMasterClient.class);
+    MasterInfo masterInfo = new MasterInfo()
+        .setMasterAddress(ADDRESS)
+        .setWebPort(WEB_PORT)
+        .setRpcPort(RPC_PORT)
+        .setStartTimeMs(START_TIME_MS)
+        .setUpTimeMs(UPTIME_MS)
+        .setVersion(VERSION)
+        .setSafeMode(SAFE_MODE);
+    Mockito.when(mMetaMasterClient.getMasterInfo(Mockito.any())).thenReturn(masterInfo);
+
+    // Prepare mock block master client
+    mBlockMasterClient = Mockito.mock(BlockMasterClient.class);
+    mCapacityBytesOnTiers.put("MEM", CAPACITY_BYTES);
+    mUsedBytesOnTiers.put("MEM", USED_BYTES);
+    BlockMasterInfo blockMasterInfo = new BlockMasterInfo()
+        .setLiveWorkerNum(LIVE_WORKER_NUM)
+        .setLostWorkerNum(LOST_WORKER_NUM)
+        .setCapacityBytes(CAPACITY_BYTES)
+        .setCapacityBytesOnTiers(mCapacityBytesOnTiers)
+        .setUsedBytes(USED_BYTES)
+        .setUsedBytesOnTiers(mUsedBytesOnTiers)
+        .setFreeBytes(FREE_BYTES);
+    Mockito.when(mBlockMasterClient.getBlockMasterInfo(Mockito.any()))
+        .thenReturn(blockMasterInfo);
+
+    // Prepare print stream
+    mOutputStream = new ByteArrayOutputStream();
+    mPrintStream = new PrintStream(mOutputStream, true, "utf-8");
+  }
+
+  @After
+  public void cleanDependencies() throws IOException {
+    mMetaMasterClient.close();
+    mBlockMasterClient.close();
+    mPrintStream.close();
+  }
+
+  @Test
+  public void callSummaryWithDependencies() throws IOException {
+    SummaryCommand summaryCommand = new SummaryCommand(mMetaMasterClient,
+        mBlockMasterClient, mPrintStream);
+    summaryCommand.run();
+    checkIfOutputValid();
+  }
+
+  /**
+   * Checks if the output is expected.
+   */
+  private void checkIfOutputValid() throws IOException {
+    String output = new String(mOutputStream.toByteArray(), StandardCharsets.UTF_8);
+    String[] outputLines = output.split("\n");
+    for (int lineNum = 0; lineNum < outputLines.length; lineNum++) {
```

Comment:
This method is very complicated. It isn't good if you need to copy the code you are testing into the test itself. Then you aren't really testing anything. I suggest doing something like
```java
List<String> expectedLines = Arrays.asList(
    "Alluxio Cluster Summary:",
    "    Master Address: testAddress",
    "    Web Port: 1231"
    ...);
```
Then we can compare the lines with a matcher designed to compare lists of strings like https://stackoverflow.com/questions/12495420/how-to-junit-test-that-two-liste-contain-the-same-elements-in-the-same-order

This way it's very clear from the test that the output is formatted how we want it. I think the test is also much easier to understand this way

Reply:
This way is much better and easy to understanding for other people.
It will also check each line and give info if not match.
A really good way! Thanks!

Changed to:
```java
    String output = new String(mOutputStream.toByteArray(), StandardCharsets.UTF_8);
    List<String> expectedOutput = Arrays.asList("Alluxio Cluster Summary: ",
        "    Master Address: testAddress",
        "    Web Port: 1231",
        ....);
    List<String> testOutput = Arrays.asList(output.split("\n"));
    Assert.assertThat(testOutput,
        IsIterableContainingInOrder.contains(expectedOutput.toArray()));
```